components/server/src/workspace/prebuild-rate-limiter.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+export type PrebuildRateLimiters = { [cloneURL: string]: PrebuildRateLimiterConfig } & {
+    "*": PrebuildRateLimiterConfig;
+};
+
+export type PrebuildRateLimiterConfig = {
+    // maximum number of requests per period
+    limit: number;
+
+    // time period which the limit is enforce against in seconds
+    period: number;
+};
+
+export namespace PrebuildRateLimiterConfig {
+    const DEFAULT_CONFIG: PrebuildRateLimiterConfig = {
+        limit: 50,
+        period: 50,
```

Comment:
Would it make sense to use `Period: 300,` as the default? Would be great to just use the default that make sense and then there is no need for additional configuration. Or do you think different installations need different rate limits?

Reply:
> Would be great to just use the default that make sense and then there is no need for additional configuration

This has been the direction when we only had self-hosted and gitpod.io, yes. When looking at Gitpod Dedicated, we need to make sure that we're in a position to quickly react to bad behavior. So far, that always turned into "quickly go to the ConfigMap and change a value". But that ad-hoc manipulation of Config/code will no longer be feasible in the near future.

I think it is important to at least pull/keep the flexibility on the installer level; slightly changing the course we were on before.

Reply:
But using the values (`period: 300`) you think is reasonable as the defaults would be good, no?

Reply:
We concluded based on a query from Sven that 100/10mins is still strong enough while less impacting valid use-cases.